composer.json

```diff
@@ -28,7 +28,7 @@
         "php": "^7.1",
         "doctrine/mongodb-odm": "^1.3 || ^2.0",
         "doctrine/mongodb-odm-bundle": "^3.0 || ^4.0",
-        "sonata-project/admin-bundle": "^3.61",
+        "sonata-project/admin-bundle": "^3.63",
```

Comment:
IMO, here is the same case than https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1017#issuecomment-604756733.

Reply:
I update the admin-bundle version in master because it's needed. But I see no reason to update it in stable branch. If someone want to stay in 3.61, he can. If someone want to update to 3.63, he still can.

So why force the update ?

Reply:
IMO, we should try to make the update first at stable branch in order to confirm there are no other required changes derived from the update.
If the update is clean and no changes are required, just the version constraint will bubble up to `master`. Otherwise, any required change will be performed first at stable branch, and then these modifications (including potential "NEXT_MEJOR" comments) will reach `master`.
I think this way prevents to change things twice and avoids possible merge conflicts.

Reply:
This is a minor update, this will need no required changes.
Plus, the stable is already working on the 3.63 since tests are running on the last version.

Reply:
Minor updates can contain changes on final classes, deprecations, etc. (which should ideally be resolved there if possible and merged here later).
I don't say the stable branch of this package is not working with sonata-project/admin-bundle:3.63, that's not my point.

Reply:
My reasoning is, if a change made here can be made first in the stable branch respecting the BC promise, is better to make the change first there (regardless it involves dependencies or not).

Reply:
I agree in generally. But not in this case.
A PR in master should not rely on some change on the stable branch.

What if you revert the 3.63 update in stable branch for any reason ? You'll introduce a bug in master. Since the change need 3.63, It's safer to update the composer.json in the same PR.

Plus this change has no benefit in stable branch

Reply:
If a difference exists between both branches, it is fine. That difference must have a reason to be, and a commit explaining why it is there. In that case, there will be no reason to not change that constraint while merging to `master`.
Please, note that I'm not thinking about a immediate benefit just for putting this constraint in `3.x`. My concern is about hypothetical scenarios which we could prevent with the path I'm proposing.
By instance, if later some other shared dependency between this package and "sonata-project/admin-bundle" updates the constraint and forces the stable branch to make changes like deprecation fixes or adding "NEXT_MAJOR" comments, these changes will be diverged between `3.x` and `master` if a similar scenario occurs first here.

Reply:
> if a similar scenario occurs first here.

Theses changes will be the problem. Not my PR.
Even if I update 3.63 in stable branch, you can still make the PR of deprecation fixes/NEXT_MAJOR in master instead of stable.

The origin of theses potential conflict will be the next potential PR, not mine. 

Plus I see no deprecation introduced in this bundle...

Reply:
>Theses changes will be the problem. Not my PR.
IMO, we should prevent potential future problems if possible (by instance, the same reasoning behind the decision to unalign multiple line assignments in order to avoid future conflicts if some other line changes).
>The origin of theses potential conflict will be the next potential PR, not mine.
If these potential conflicts occur, they are no potential anymore. Avoiding them is a continuous preventive work which must be done precisely to avoid them.

Reply:
> By instance, the same reasoning behind the decision to unalign multiple line assignments in order to avoid future conflicts if some other line changes).

There is ATM no coding-standard check about this so this is just your point of view. Others can explained that alignement help readability, and readability help to make less mistakes.

> > The origin of theses potential conflict will be the next potential PR, not mine.
> 
> If these potential conflicts occur, they are no potential anymore. Avoiding them is a continuous preventive work which must be done precisely to avoid them.

But this avoid nothing ! Even if we update or not 3.63 in master or in stable. I can still make a PR to remove deprecation in master or stable and create conflict or not. There is no link.

I will end by doing it because it's make stupidly loosing time for such a tiny PR. But I don't see why this is triggering you when you and others contributors approved this or others similar PR.

Reply:
>There is ATM no coding-standard check about this so this is just your point of view.

Yes, my review and my proposals are based on my opinionated point of view.

>I will end by doing it because it's make stupidly loosing time for such a tiny PR. But I don't see why this is triggering you when you and others contributors approved this or others similar PR.

It's just a proposal, even, a request. If you think this can't or mustn't be done in this way you can disagree with my opinion and that is fine too. Nobody has the absolute truth.
I think that's why the review process exists.
It's very possible that I've approved PRs like this before, even, that I've created PRs like this myself. That's because my review is tied to my own criteria and the needs I have closer.

Reply:
@phansys it's not a big deal to have this conflict. I'd rather the contributors have 1 PR to do for cases such as this where bumping the dependency alone on the stable branch does not have a justification by itself.